pkg/kustomize/kustomize_cli.go

```diff
@@ -0,0 +1,40 @@
+package kustomize
+
+import (
+	"github.com/jenkins-x/jx/pkg/util"
+	"regexp"
+)
+
+// KustomizeCLI implements common kustomize actions based on kustomize CLI
+type KustomizeCLI struct {
+	Runner util.Commander
+}
+
+// NewKustomizeCLI creates a new KustomizeCLI instance configured to use the provided kustomize CLI in
+// the given current working directory
+func NewKustomizeCLI() *KustomizeCLI {
+	runner := &util.Command{
+		Name: "kustomize",
+	}
+	cli := &KustomizeCLI{
+		Runner: runner,
+	}
+	return cli
+}
+
+// Version executes the Kustomize version command and returns its output
+func (k *KustomizeCLI) Version(extraArgs ...string) (string, error) {
+	args := []string{"version", "--short"}
+	args = append(args, extraArgs...)
+	version, err := k.runKustomizeWithOutput(args...)
+	if err != nil {
+		return "", err
+	}
+	regex := regexp.MustCompile(`[0-9]+\.[0-9]+\.[0-9]+`)
```

Comment:
I'd extract the regular expression stuff into a function like:

`extractSemanticVersion(fullVersionString string) (string, error)` 

The main reason is that you can add some simple tests to ensure that this regular expression handles the version strings we know about properly. It also serves as documentation on what we expect to receive by the version call to `customize`.